src/main/java/com/onevizion/scmdb/vo/SqlScript.java

```diff
@@ -177,6 +178,15 @@ public boolean equals(Object o) {
 
     @Override
     public int compareTo(SqlScript anotherScript) {
-        return name.compareTo(anotherScript.getName());
+        return extractNumber(name) - extractNumber(anotherScript.getName());
```

Comment:
Create `number` field and fill it inside `create()` method, then use it in `create`  instead of extracting script number on each method call